src/tests/multihost/alltests/test_proxy.py

```diff
@@ -80,6 +80,67 @@ def test_0001_1724717(self, multihost):
         multihost.client[0].run_command(del_user)
         multihost.client[0].run_command(restore)
 
+    def test_0003_update_removed_grp_membership(self, multihost,
+                                                backupsssdconf):
+        """
+        :title: proxy: secondary group is shown in sssd cache after
+         group is removed
+        :id: 7cfb9aa9-6e68-4914-afb8-ecfae132aa84
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1917970
+        :customerscenario: true
+        :steps:
+          1. Edit sssd.conf and configure proxy provider with
+             entry_cache_timeout = 1
+          2. Restart SSSD with cleared cache
+          3. Create a localuser and localgroup
+          4. Add that localuser to the localgroup
+          5. Assert localgroup is shown in localuser's group list
+          6. Remove localuser from localgroup
+          7. Assert that after entry_cache_timeout, localuser's groups
+             are not listing localgroup
+        :expectedresults:
+          1. Should succeed
+          2. Should succeed
+          3. Should succeed
+          4. Should succeed
+          5. Should succeed
+          6. Should succeed
+          7. Should succeed
+        """
+        tools = sssdTools(multihost.client[0])
+        domain_name = tools.get_domain_section_name()
+        multihost.client[0].service_sssd('stop')
+        tools.remove_sss_cache('/var/lib/sss/db')
+        l_usr, l_grp = 'testuser', 'testgroup'
+        multihost.client[0].run_command(f'useradd {l_usr}')
+        multihost.client[0].run_command(f'groupadd {l_grp}')
+        multihost.client[0].run_command(f'usermod -aG {l_grp} {l_usr}')
+        domain_params = {'id_provider': 'proxy',
+                         'proxy_lib_name': 'files',
+                         'auth_provider': 'krb5',
+                         'ignore_group_members': 'False',
+                         'cache_credentials': 'True',
+                         'entry_cache_timeout': '1',
+                         'krb5_validate': 'True'}
+        tools.sssd_conf('domain/%s' % domain_name, domain_params)
+        del_domain_params = {'ldap_uri': 'ldaps:%s' %
+                             (multihost.master[0].run_command),
+                             'ldap_tls_cacert':
+                             '/etc/openldap/cacerts/cacert.pem',
+                             'ldap_search_base': ds_suffix,
+                             'use_fully_qualified_names': 'True'}
+        tools.sssd_conf('domain/%s' % domain_name,
+                        del_domain_params, action='delete')
+        multihost.client[0].service_sssd('start')
+        cmd = multihost.client[0].run_command(f'groups {l_usr}')
+        assert 'testgroup' in cmd.stdout_text
+        multihost.client[0].run_command(f'gpasswd -d {l_usr} {l_grp}')
+        time.sleep(1)
```

Comment:
Will it be better to keep entry cache timeout higher (10)? Check that testgroup is still there after a few seconds (sleep 5), then after a few more seconds of sleep (sleep 5) testgroup is not there?

Reply:
Increasing cache timeout is not going to help. Cache must get updated once cache timeout is reached.

Reply:
Checking that the group is returned even after it is removed shows that it exists in cache and then after the timeout the cache is cleared.

Reply:
The problematic behavior in this bug is not because of entry_cache_timeout. From the bug comments, seemingly the issue is in the proxy_provider which was not updating memberof. That is why, in previous version, removed group was returned even after entry_cache_timeout was reached. The customer in the bug-description has categorically mentioned that using entry_cache_timeout option had not helped to get correct behavior.
The fixed version shows correct behavior of the changes in group membership reflected correctly.